scripts/ci/assay_runner_lane_check.py

```
@@ -1743,6 +1744,53 @@ def self_test() -> None:
+    never yields a directory.
+
+    Bounded on purpose: this proves no declared surface is dead. It does not
+    prove the gated set is the right set. Most gating lives in `classify_file`
+    rather than the manifest, and which of the two is the source of truth is
+    open in #2020.
+    """
+    config = load_gated_path_config()
+    root = Path(__file__).resolve().parents[2]
+    tracked = {
+        path
+        for path in subprocess.check_output(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)